generic/cxl.py

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2017 IBM
+# Author: sudeesh john <sudeesh@linux.vnet.ibm.com>
+
+import os
+
+from avocado import Test, main
+from avocado.utils import build, git, process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Cxl(Test):
+    """
+        This tests the CAPI functionality in IBM Power machines.
+        This wrapper uses the testcases from
+        https://github.com/ibm-capi/cxl-tests.git
+    """
+
+    def setUp(self):
+        """
+            Preparing the machie for the cxl test
+        """
+        self.script = self.params.get('script', default='memcpy_afu_ctx')
+        self.args = self.params.get('args', default='')
+        lspci_out = process.system_output("lspci")
+        if "accelerators" not in lspci_out:
+            self.cancel("No capi card preset. Unable to initialte the test")
+        smngr = SoftwareManager()
+        deps = ['gcc', 'make', 'automake', 'autoconf']
+        for pkgs in deps:
+            if not smngr.check_installed(pkgs) and not smngr.install(pkgs):
+                self.error(pkgs + ' is needed for the test to be run')
+        git.get_repo('https://github.com/ibm-capi/cxl-tests.git',
```

Comment:
use `https://github.com/ibm-capi/cxl-tests/archive/master.zip` for download URL that help to use fetch_asset lib as that lib has more functionality

Reply:
I think I didn't understand this. Whats the extra benefit that I can use in this test ?

Reply:
fetch_asset we have a param as expire which used to avoid download again as till when time got expire but in git case it will do every time i think that is not needed